src/Fetch.js

```diff
@@ -585,3 +585,12 @@ function _fetch_get_response_headers(id, dst, dstSizeBytes) {
     stringToUTF8(responseHeaders, dst, dstSizeBytes);
     return Math.min(lengthBytes, dstSizeBytes);
 }
+
+//Delete the xhr JS object, allowing it to be garbage collected.
+function _fetch_free_xhrs(id) {
+  //Note: should just be [id], but indexes off by 1 (see: #8803)
+#if FETCH_DEBUG
+  console.log("fetch: Deleting id:" + (id-1) + " of " + Fetch.xhrs.length);
+#endif
+  Fetch.xhrs.splice(id-1, 1);
```

Comment:
Oh hmm, `.splice` won't work here since that will shift all the subsequent indexes down by one, hence the fetch structures will point to incorrect indices. The only way to go about here is to do `Fetch.xhrs[id-1] = null;`. (that will generate a really slow running leak as the number of `null`s grow in the list, but best way to fix that in the future is to move all JS objects to a common global table, which can then warrant maintaining a general index reuse algorithm to avoid memory leaks)

Also, let's rename `_fetch_free_xhrs` to just `_fetch_free`, perhaps there will be other things to free up in a fetch on JS side in the future, in which case more appropriate to call it a generic free function.

Reply:
Ugh, you're right. good catch. I've changed it to `delete Fetch.xhrs[id-1]`.
So I guess this doesn't completely remove the leak, just greatly reduce it.